Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/Cart.kt

```diff
@@ -16,20 +16,323 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.ChainStyle
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ConstraintLayout
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.preferredHeight
+import androidx.compose.foundation.layout.preferredHeightIn
+import androidx.compose.foundation.layout.preferredSize
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.foundation.lazy.ExperimentalLazyDsl
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.text.LastBaseline
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.RectangleShape
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.ui.tooling.preview.Preview
 import com.example.jetsnack.R
+import com.example.jetsnack.model.Snack
+import com.example.jetsnack.model.SnackCollection
+import com.example.jetsnack.model.SnackRepo
+import com.example.jetsnack.ui.components.JetsnackButton
+import com.example.jetsnack.ui.components.JetsnackDivider
+import com.example.jetsnack.ui.components.JetsnackSurface
+import com.example.jetsnack.ui.components.QuantitySelector
+import com.example.jetsnack.ui.components.SnackCollection
+import com.example.jetsnack.ui.components.SnackImage
+import com.example.jetsnack.ui.theme.AlphaNearOpaque
+import com.example.jetsnack.ui.theme.JetsnackTheme
+import com.example.jetsnack.ui.utils.statusBarsHeight
 
 @Composable
-fun Cart(modifier: Modifier = Modifier) {
-    Text(
-        text = stringResource(R.string.home_cart),
+fun Cart(
+    onSnackClick: (Long) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val cartSnacks = remember { SnackRepo.getCart() }
+    val inspiredByCart = remember { SnackRepo.getInspiredByCart() }
+    Cart(cartSnacks, inspiredByCart, onSnackClick, modifier)
+}
+
+@Composable
+fun Cart(
+    cartSnacks: List<Snack>,
+    inspiredByCart: SnackCollection,
+    onSnackClick: (Long) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    JetsnackSurface(modifier = modifier.fillMaxSize()) {
+        Box {
```

Comment:
Feels a bit strange to use a Box and not provide any modifiers to the children (other than the bottom one)

Reply:
`Box` acts here as what `Stack` used to be, right? I think it's fine.

Reply:
Said that, I think `CheckoutBar` shouldn't be included in here as it's not drawn on top of `CartContent`

Reply:
For now, as there's only one button in the checkout bar, I actually do draw it on top of `CartContent` (with a tiny bit of transparency so you can see what's behind it).